[IMP] inventory: visibility days (#7995)
odoo/documentation: content/applications/inventory_and_mrp/inventory/product_management/product_replenishment/reordering_rules.rst

Create new reordering rules
===========================

To create a new reordering rule, navigate to :menuselection:`Inventory app --> Configuration -->
Reordering Rules`, then click :guilabel:`New`, and fill out the new line as follows:

- :guilabel:`Product`: The product that is replenished by the rule.

> - :guilabel:`Location`: The location where the product is stored.

Author: I left it out, because it has its own dedicated section later in the doc (:

   Operations --> Replenishment`.

To enable the :guilabel:`Trigger` field, go to :menuselection:`Inventory app --> Configuration -->
Reordering Rules`. Then, click the :guilabel:`(slider)` icon, located to the far-right of the column
titles, and enable the :guilabel:`Trigger` option from the additional options drop-down menu that
appears.

.. image:: reordering_rules/enable-trigger.png
   :align: center
   :alt: Enable the Trigger field by toggling it in the additional options menu.

In the :guilabel:`Trigger` column, select :guilabel:`Auto` or :guilabel:`Manual`. Refer to the
sections below to learn about the different types of reordering rules.

Auto
----

Automatic reordering rules, configured by setting the reordering rule's :guilabel:`Trigger` field to
:guilabel:`Auto`, generates purchase or manufacturing orders when:

#. the scheduler runs, and the *On Hand* quantity is below the minimum
#. a sales order is confirmed, and lowers the *Forecasted* quantity of the product below the minimum

.. tip::
   The scheduler is set to run once a day, by default.

   To manually trigger a reordering rule before the scheduler runs, ensure :ref:`developer mode

>    <developer-mode>` is enabled, select :menuselection:`Inventory app --> Operations --> Run

Reviewer: ```suggestion
   <developer-mode>` is enabled, and then select :menuselection:`Inventory app --> Operations --> Run
```

.. example::
   The product, `Office Lamp`, has an automatic reordering rule set to trigger when the forecasted
   quantity falls below the :guilabel:`Min Quantity` of `5.00`. Since the current
   :guilabel:`Forecast` is `55.00`, the reordering rule is **not** triggered.

   .. image:: reordering_rules/auto.png
      :align: center
      :alt: Show automatic reordering rule from the Reordering Rule page.

If the :guilabel:`Buy` route is selected, then an :abbr:`RFQ (Request for Quotation)` is generated.
To view and manage :abbr:`RFQs (Requests for Quotation)`, navigate to :menuselection:`Purchase app
--> Orders --> Requests for Quotation`.

If the :guilabel:`Manufacture` route is selected, then an :abbr:`MO (Manufacturing Order)` is
generated. To view and manage :abbr:`MOs (Manufacturing Orders)`, navigate to
:menuselection:`Manufacturing app --> Operations --> Manufacturing Orders`.

When no route is selected, Odoo selects the :guilabel:`Route` specified in the :guilabel:`Inventory`
tab of the product form.

.. _inventory/product_management/manual-rr:

Manual
------

Manual reordering rules, configured by setting the reordering rule's :guilabel:`Trigger` field to

> :guilabel:`Manual`, lists a product on the replenishment dashboard when the forecasted quantity

Reviewer: ```suggestion
:guilabel:`Manual`, list a product on the replenishment dashboard when the forecasted quantity
```

falls below a specified minimum. Products on this dashboard are called *needs*, because they are

> needed to fulfill upcoming sales orders, but the forecasted quantity is not enough.

Reviewer: ```suggestion
needed to fulfill upcoming sales orders, for which the forecasted quantity is not enough.
```

(:menuselection:`Inventory app --> Operations --> Replenishment`) a certain number of days in
advance.

.. example::
   A product has a manual reordering rule set to trigger when the stock level falls below four
   units. The current on-hand quantity is ten units.

   The current date is February 20th, and the *delivery date* on a sales order (in the
   :guilabel:`Other Info` tab) is March 3rd — twelve days from the current date.

   The :ref:`vendor lead time <inventory/management/purchase-lt>` is four days, and the
   :ref:`purchase security lead time <inventory/management/purchase-security-lt>` is one day.

   When the :guilabel:`Visibility Days` field of the reordering rule is set to zero, the product
   appears on the replenishment dashboard five days before the delivery date, which, in this case,
   is February 27th.

   .. image:: reordering_rules/need-dates.png
      :align: center
      :alt: Graphic representing when the need appears on the replenishment dashboard: Feb 27th.

   To see the product on the replenishment dashboard for the current date, February 20, set the
   :guilabel:`Visibility Days` to `7.00`.

> To determine the amount of visibility days needed to see a product on the replenishment dashboard,

Reviewer: ```suggestion
To determine the number of visibility days needed to see a product on the replenishment dashboard,
```

.. tip::
   Reordering rules can also be created from each product form. To do so, navigate to
   :menuselection:`Inventory app --> Products --> Products`, then select a product. Click on
   :menuselection:`Reordering Rules --> New`, then fill out the new line, as detailed above.

For advanced usage of reordering rules, learn about the following reordering rule fields:

> 

Author: Addressed this in the note below (:

- :ref:`Trigger <inventory/product_management/trigger>`
- :ref:`Visibility days <inventory/product_management/visibility-days>`
- :ref:`Preferred route <inventory/product_management/route>`

.. _inventory/product_management/trigger:

Trigger
=======

When stock falls below the reordering rule's minimum, set the reordering rule's *trigger* to
*automatic* to automatically create purchase or manufacturing orders to replenish stock.

Alternatively, setting the reordering rule's trigger to *manual* displays the product and forecasted
stock on the *replenishment dashboard*, where the procurement manager can review the stock levels,
lead times, and forecasted dates of arrival.

.. seealso::
   :doc:`strategies`

.. tip::

>    The replenishment dashboard is accessible by going to :menuselection:`Inventory app -->

Reviewer 2: ```suggestion
   The :guilabel:`Replenishment` dashboard is accessible by going to :menuselection:`Inventory app -->
```
I am suggesting the use of a GUI here because, once they use the provided directions to navigate to that specific page, the actual title of 'Replenishment' *does*, in fact, appear on the UI/screen at that time. So, that's my reasoning behind the suggestion. Feel free to take or leave (if you disagree with that logic) -- totally up to you 👍

Author: This logic makes sense to me-- thanks for the suggestion!

   Operations --> Replenishment`.

To enable the :guilabel:`Trigger` field, go to :menuselection:`Inventory app --> Configuration -->

> Reordering Rules`. Then, click the :guilabel:`(slider)` icon, located to the far-right of the column

Reviewer 2: ah, I see that you *do* mention the slider icon...feel free to disregard my previous comment about it

> titles, and enable the :guilabel:`Trigger` option from the additional options drop-down menu that

Reviewer 2: ```suggestion
titles, represented by :guilabel:`two horizontal lines with dots on them`, and enable the :guilabel:`Trigger` option from the additional options drop-down menu that
```
my suggested pitch for how to visually describe the slider icon, in lieu of an accurate emoji

Author: I personally would rather leave the slider icon concise and not described, while circling it in my screenshot right below. (: Thank you for bringing it up, though! I appreciate the insight on how you describe that icon.

appears.

.. image:: reordering_rules/enable-trigger.png
   :align: center
   :alt: Enable the Trigger field by toggling it in the additional options menu.

In the :guilabel:`Trigger` column, select :guilabel:`Auto` or :guilabel:`Manual`. Refer to the
sections below to learn about the different types of reordering rules.

Auto
----

Automatic reordering rules, enabled by setting the reordering rule's :guilabel:`Trigger` field to
:guilabel:`Auto`, generate purchase or manufacturing orders when:

#. the scheduler runs, and the *On Hand* quantity is below the minimum
#. a sales order is confirmed, and lowers the *Forecasted* quantity of the product below the minimum

.. tip::
   The scheduler is set to run once a day, by default.

   To manually trigger a reordering rule before the scheduler runs, ensure :ref:`developer mode
   <developer-mode>` is enabled, and then select :menuselection:`Inventory app --> Operations -->

>    Run Scheduler`. Then, select the green :guilabel:`Run Scheduler` button on the pop-up that

Reviewer 2: ```suggestion
   Run Scheduler`. Then, select the green :guilabel:`Run Scheduler` button on the pop-up window that
```

   appears.

>    Be aware that this also triggers any other scheduled actions.

Reviewer 2: ```suggestion
   Be aware that this also triggers *any other* scheduled actions.
```

When :ref:`manual reordering rules <inventory/product_management/manual-rr>` are assigned to a
product, *visibility days* make the product appear on the replenishment dashboard
(:menuselection:`Inventory app --> Operations --> Replenishment`) a certain number of days in
advance.

.. example::
   A product has a manual reordering rule set to trigger when the stock level falls below four
   units. The current on-hand quantity is ten units.

   The current date is February 20th, and the *delivery date* on a sales order (in the
   :guilabel:`Other Info` tab) is March 3rd — twelve days from the current date.

   The :ref:`vendor lead time <inventory/management/purchase-lt>` is four days, and the
   :ref:`purchase security lead time <inventory/management/purchase-security-lt>` is one day.

   When the :guilabel:`Visibility Days` field of the reordering rule is set to zero, the product
   appears on the replenishment dashboard five days before the delivery date, which, in this case,
   is February 27th.

   .. image:: reordering_rules/need-dates.png
      :align: center
      :alt: Graphic representing when the need appears on the replenishment dashboard: Feb 27th.

>    To see the product on the replenishment dashboard for the current date, February 20, set the

Reviewer 2: ```suggestion
   To see the product on the replenishment dashboard for the current date, February 20th, set the
```

.. example::
   Referring to the example above, today's date is February 20th, and the need for the product
   appears on February 27th.

   (February 27 - February 20 = 7 days)

   Incorrectly setting the :guilabel:`Visibility Days` fewer than seven days in this case results in
   the need **not** appearing on the replenishment dashboard.

   .. image:: reordering_rules/visibility-days.png
      :align: center
      :alt: Show the replenishment dashboard with the correct and incorrect visibility days set.

.. _inventory/product_management/route:

Preferred route
===============

Odoo allows for multiple routes to be selected under the :guilabel:`Inventory` tab on each product
form. For instance, it is possible to select both :guilabel:`Buy` and :guilabel:`Manufacture`, thus
enabling the functionality of both routes.

Odoo also enables users to set a preferred route for a product's reordering rule. This is the route
that the rule defaults to if multiple are selected. To select a preferred route, begin by navigating
to :menuselection:`Inventory app --> Configuration --> Reordering Rules` or
:menuselection:`Inventory app --> Operations --> Replenishment`.

> Click inside of the column on the row of a reordering rule and a drop-down menu shows all available

Reviewer 2: ```suggestion
Click inside of the column on the row of a reordering rule, and a drop-down menu shows all available
```
